eth/precompiles/modexp.py

```diff
@@ -121,18 +126,19 @@ def _modexp(data: bytes) -> int:
     return result
 
 
-def modexp(computation: BaseComputation) -> BaseComputation:
+@curry
+def modexp(computation: BaseComputation,
+           gas_cost: Callable[..., int] = _compute_modexp_gas_fee_eip_198) -> BaseComputation:
     """
     https://github.com/ethereum/EIPs/pull/198
     """
     data = computation.msg.data_as_bytes
 
-    gas_fee = _compute_modexp_gas_fee(data)
-    computation.consume_gas(gas_fee, reason='MODEXP Precompile')
+    computation.consume_gas(gas_cost(data), reason='MODEXP Precompile')
```

Comment:
*nit* I liked these on separate lines, but if you disagree, no need to block the PR over it. :)